spin fails to build from source with GCC 15 on amd64

The package spin with version 3.2.0-r2 fails to build from source with GCC 15 on amd64 with the following error:

```2025/05/31 00:55:00 WARN   cargo:warning=  232 | #define st_cleanup_safe             onig_st_cleanup_safe
2025/05/31 00:55:00 WARN   cargo:warning=      |                                     ^~~~~~~~~~~~~~~~~~~~
2025/05/31 00:55:00 WARN   cargo:warning=oniguruma/src/st.c:477:1: note: in expansion of macro 'st_cleanup_safe'
2025/05/31 00:55:00 WARN   cargo:warning=  477 | st_cleanup_safe(table, never)
2025/05/31 00:55:00 WARN   cargo:warning=      | ^~~~~~~~~~~~~~~
2025/05/31 00:55:00 WARN   cargo:warning=oniguruma/src/st.c:483:21: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
2025/05/31 00:55:00 WARN   cargo:warning=  483 |   st_foreach(table, delete_never, never);
2025/05/31 00:55:00 WARN   cargo:warning=      |                     ^~~~~~~~~~~~
2025/05/31 00:55:00 WARN   cargo:warning=      |                     |
2025/05/31 00:55:00 WARN   cargo:warning=      |                     int (*)(st_data_t,  st_data_t,  st_data_t) {aka int (*)(long unsigned int,  long unsigned int,  long unsigned int)}
2025/05/31 00:55:00 WARN   cargo:warning=oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(st_data_t,  st_data_t,  st_data_t)' {aka 'int (*)(long unsigned int,  long unsigned int,  long unsigned int)'}
2025/05/31 00:55:00 WARN   cargo:warning=   55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
2025/05/31 00:55:00 WARN   cargo:warning=      |                               ^~~~~~~~~~~~~~~~
2025/05/31 00:55:00 WARN   cargo:warning=oniguruma/src/st.h:35:18: note: in definition of macro '_'
2025/05/31 00:55:00 WARN   cargo:warning=   35 | # define _(args) args
2025/05/31 00:55:00 WARN   cargo:warning=      |                  ^~~~
2025/05/31 00:55:00 WARN   cargo:warning=oniguruma/src/st.c:465:1: note: 'delete_never' declared here
2025/05/31 00:55:00 WARN   cargo:warning=  465 | delete_never(st_data_t key __attribute__ ((unused)), st_data_t value,
2025/05/31 00:55:00 WARN   cargo:warning=      | ^~~~~~~~~~~~
2025/05/31 00:55:00 WARN   cargo:warning=oniguruma/src/st.c: In function 'onig_st_foreach':
2025/05/31 00:55:00 WARN   cargo:warning=oniguruma/src/regint.h:229:37: warning: old-style function definition [-Wold-style-definition]
2025/05/31 00:55:00 WARN   cargo:warning=  229 | #define st_foreach                  onig_st_foreach
2025/05/31 00:55:00 WARN   cargo:warning=      |                                     ^~~~~~~~~~~~~~~
2025/05/31 00:55:00 WARN   cargo:warning=oniguruma/src/st.c:488:1: note: in expansion of macro 'st_foreach'
2025/05/31 00:55:00 WARN   cargo:warning=  488 | st_foreach(table, func, arg)
2025/05/31 00:55:00 WARN   cargo:warning=      | ^~~~~~~~~~
2025/05/31 00:55:00 WARN   cargo:warning=oniguruma/src/st.c:500:23: error: too many arguments to function 'func'; expected 0, have 3
2025/05/31 00:55:00 WARN   cargo:warning=  500 |             retval = (*func)(ptr->key, ptr->record, arg);
2025/05/31 00:55:00 WARN   cargo:warning=      |                      ~^~~~~~ ~~~~~~~~
2025/05/31 00:55:00 WARN   cargo:warning=oniguruma/src/st.c: In function 'strhash':
2025/05/31 00:55:00 WARN   cargo:warning=oniguruma/src/st.c:538:1: warning: old-style function definition [-Wold-style-definition]
2025/05/31 00:55:00 WARN   cargo:warning=  538 | strhash(string)
2025/05/31 00:55:00 WARN   cargo:warning=      | ^~~~~~~
2025/05/31 00:55:00 WARN   cargo:warning=oniguruma/src/st.c: In function 'numcmp':
2025/05/31 00:55:00 WARN   cargo:warning=oniguruma/src/st.c:577:1: warning: old-style function definition [-Wold-style-definition]
2025/05/31 00:55:00 WARN   cargo:warning=  577 | numcmp(x, y)
2025/05/31 00:55:00 WARN   cargo:warning=      | ^~~~~~
2025/05/31 00:55:00 WARN   cargo:warning=oniguruma/src/st.c: In function 'numhash':
2025/05/31 00:55:00 WARN   cargo:warning=oniguruma/src/st.c:584:1: warning: old-style function definition [-Wold-style-definition]
2025/05/31 00:55:00 WARN   cargo:warning=  584 | numhash(n)
2025/05/31 00:55:00 WARN   cargo:warning=      | ^~~~~~~
2025/05/31 00:55:00 WARN
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 1
2025/05/31 00:55:00 WARN   cargo:warning=ToolExecError: Command "cc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "/home/build/target/release/build/onig_sys-e75fa2d74bcc9e75/out" "-I" "oniguruma/src" "-DHAVE_UNISTD_H=1" "-DHAVE_SYS_TYPES_H=1" "-DHAVE_SYS_TIME_H=1" "-o" "/home/build/target/release/build/onig_sys-e75fa2d74bcc9e75/out/c77b18e714869709-st.o" "-c" "oniguruma/src/st.c" with args cc did not execute successfully (status code exit status: 1).
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN   exit status: 0
2025/05/31 00:55:00 WARN
2025/05/31 00:55:00 WARN   --- stderr
2025/05/31 00:55:00 WARN
2025/05/31 00:55:00 WARN
2025/05/31 00:55:00 WARN   error occurred: Command "cc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "/home/build/target/release/build/onig_sys-e75fa2d74bcc9e75/out" "-I" "oniguruma/src" "-DHAVE_UNISTD_H=1" "-DHAVE_SYS_TYPES_H=1" "-DHAVE_SYS_TIME_H=1" "-o" "/home/build/target/release/build/onig_sys-e75fa2d74bcc9e75/out/c77b18e714869709-st.o" "-c" "oniguruma/src/st.c" with args cc did not execute successfully (status code exit status: 1).
2025/05/31 00:55:00 WARN
2025/05/31 00:55:00 WARN
2025/05/31 00:55:00 WARN warning: build failed, waiting for other jobs to finish...
2025/05/31 00:56:09 ERRO failed to build package: unable to run package spin pipeline: unable to run pipeline: exit status 101
make[1]: *** [Makefile:157: packages/x86_64/spin-3.2.0-r2.apk] Error 1
make[1]: Leaving directory '/home/sergio_durigan_chainguard_dev/os'
make: *** [Makefile:151: package/spin] Error 2
```

--- a/bat.yaml
+++ b/bat.yaml
@@ -1,7 +1,7 @@
 package:
   name: bat
   version: 0.25.0
-  epoch: 2
+  epoch: 3
   description: "A cat(1) clone with wings"
   copyright:
     - license: MIT OR Apache-2.0
@@ -13,7 +13,6 @@ environment:
       - busybox
       - ca-certificates-bundle
       - cargo-auditable
-      - gcc-14-default
       - rust
       - zlib-dev
 

--- a/bat/cargobump-deps.yaml
+++ b/bat/cargobump-deps.yaml
@@ -9,3 +9,6 @@ packages:
   # Newer version of url pulls in newer version of idna necesary to fix GHSA-h97m-ww89-6jmq CVE
   - name: url
     version: 2.5.4
+  # fix build error with gcc 15
+  - name: onig
+    version: 6.5.1

--- a/delta.yaml
+++ b/delta.yaml
@@ -1,7 +1,7 @@
 package:
   name: delta
   version: 0.18.2
-  epoch: 2
+  epoch: 3
   description: Syntax-highlighting pager for git and diff output
   copyright:
     - license: MIT
@@ -12,7 +12,6 @@ environment:
       - build-base
       - busybox
       - ca-certificates-bundle
-      - gcc-14-default
       - rust
 
 pipeline:
@@ -21,6 +20,8 @@ pipeline:
       repository: https://github.com/dandavison/delta
       tag: ${{package.version}}
       expected-commit: a589ff9debaefdd3992384434120f5a03a103481
+
+  - uses: rust/cargobump
 
   - runs: |
       cargo build --release

--- a/delta/cargobump-deps.yaml
+++ b/delta/cargobump-deps.yaml
@@ -1,0 +1,5 @@
+packages:
+    # fix build error with gcc 15
+    # https://github.com/dandavison/delta/issues/1990
+    - name: onig
+      version: 6.5.1

--- a/kdash.yaml
+++ b/kdash.yaml
@@ -1,7 +1,7 @@
 package:
   name: kdash
   version: "0.6.2"
-  epoch: 5
+  epoch: 6
   description: "A simple and fast dashboard for Kubernetes"
   copyright:
     - license: MIT
@@ -13,7 +13,6 @@ environment:
       - busybox
       - ca-certificates-bundle
       - cargo-auditable
-      - gcc-14-default
       - libxcb
       - libxcb-dev
       - openssl

--- a/kdash/cargobump-deps.yaml
+++ b/kdash/cargobump-deps.yaml
@@ -11,3 +11,5 @@ packages:
       version: 0.17.13
     - name: tokio
       version: 1.43.1
+    - name: onig
+      version: 6.5.1

--- a/oranda.yaml
+++ b/oranda.yaml
@@ -1,7 +1,7 @@
 package:
   name: oranda
   version: 0.6.5
-  epoch: 7
+  epoch: 8
   description: generate beautiful landing pages for your developer tools
   copyright:
     - license: MIT
@@ -13,7 +13,6 @@ environment:
       - busybox
       - ca-certificates-bundle
       - cargo-auditable
-      - gcc-14-default
       - openssl-dev
       - rust
 

--- a/oranda/cargobump-deps.yaml
+++ b/oranda/cargobump-deps.yaml
@@ -7,3 +7,5 @@ packages:
       version: 1.38.2
     - name: crossbeam-channel
       version: 0.5.15
+    - name: onig
+      version: 6.5.1

--- a/spin.yaml
+++ b/spin.yaml
@@ -1,7 +1,7 @@
 package:
   name: spin
   version: "3.2.0"
-  epoch: 2
+  epoch: 3
   description: "Spin is the open source developer tool for building and running serverless applications powered by WebAssembly."
   copyright:
     - license: Apache-2.0
@@ -19,8 +19,8 @@ environment:
       - build-base
       - busybox
       - ca-certificates-bundle
+      - cargobump
       - cmake
-      - gcc-14-default
       - openssl-dev
       - rustup
       - wolfi-base
@@ -53,6 +53,8 @@ pipeline:
           echo '[build]' >> .cargo/config
           echo 'rustflags = ["-Ctarget-feature=+fp16"]' >> .cargo/config
       fi
+
+      cargobump --run-update=false --bump-file ./cargobump-deps.yaml
 
       cargo build --release
 

--- a/spin/cargobump-deps.yaml
+++ b/spin/cargobump-deps.yaml
@@ -1,0 +1,6 @@
+packages:
+    # fix build error with gcc 15
+    # https://github.com/spinframework/spin/issues/3128
+    - name: onig
+      version: 6.5.1
+

